site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md

```diff
@@ -0,0 +1,403 @@
+    name: IDPs on FederationDomains
+    weight: 20
+    parent: howto-configure-supervisor
+---
+
+This guide explains how to associate one or more external identity providers (IDPs) with a FederationDomain.
+It also details how to configure identity transformations and identity policies for those identity
+providers.
+
+## Prerequisites
+
+This how-to guide assumes that you have already [installed the Pinniped Supervisor]({{< ref "install-supervisor" >}})
+and have already read the guide about how to [configure the Supervisor as an OIDC issuer]({{< ref "configure-supervisor" >}}).
+
+This guide focuses on the use of the `spec.identityProviders` setting on the FederationDomain CRD.
+
+## Summary
+
+External identity providers may be configured in the Supervisor by creating OIDCIdentityProvider,
+ActiveDirectoryIdentityProvider, or LDAPIdentityProvider resources in the same namespace as the Supervisor.
+
+There are two ways to configure which of these external identity providers shall be used by a FederationDomain.
+
+1. When there is no `spec.identityProviders` configured on a FederationDomain, then the FederationDomain will use
+   the one and only identity provider that is configured in the same namespace. This provides backwards compatibility
+   with older configurations of Supervisors from before the `spec.identityProviders` setting was added to the
+   FederationDomain CRD. There must be exactly one OIDCIdentityProvider,
+   ActiveDirectoryIdentityProvider, or LDAPIdentityProvider resource in the same namespace as the Supervisor.
+   If there are no identity provider resources, or if there are more than one, then the FederationDomain will
+   not allow any users to authenticate, and a error message will be shown in its `status`.
+
+2. When `spec.identityProviders` is explicitly configured on a FederationDomain, then the FederationDomain will
+   allow clients to use any of those identity providers to authenticate. In this case, you may optionally also configure
+   identity transformations and policies that the FederationDomain should apply to each of these identity providers
+   (see below for details). When using the `pinniped get kubeconfig` CLI command, you will need to choose for
+   which identity provider you would like to generate a kubeconfig. A cluster may have multiple kubeconfigs,
+   e.g. one for each identity provider.
+
+The remainder of this guide focuses on the second case, and describes the settings that may be used to explicitly
+configure which identity providers are used, along with optional identity transformations and policies.
+
+## Configuring a FederationDomain's identity providers
+
+A user may authenticate to a FederationDomain using any of the IDPs configured in the FederationDomain's
+`spec.identityProviders`. To add IDPs to this list, simply configure each as a reference to the type and name
+of the resource. The identity provider resources must be in the same namespace where the Supervisor was installed.
```

Comment:
Should we link to `docs/howto/install-supervisor/`?

Reply:
Updated